mailheader.js

```diff
@@ -201,7 +197,7 @@ function _decode_header (matched, encoding, lang, cte, data) {
 
     switch (cte) {
         case 'Q':
-            data = utils.decode_qp(data.replace(/_/g, ' '));
+            data = libqp.decode(data.replace(/_/g, ' '));
```

Comment:
since this is in a header, why are we using libqp.decode instead of libmime.decodeWords?

Reply:
The function libmime.decodeWords works on the whole encoded-word thing, like "=?iso-8859-1?q?this=20is=20some=20text?=".
This deep into the code we're only looking at the part between the last two '?' chars, like "this=20is=20some=20text", so the decoding will work (correctly) the same way as quoted-printable at this point.
I considered re-working more of the code in mailheader.js to use the functionality of libmime, but decided against doing so as I'm not aware (right now) of any actual bugs in the mailheader.js version; so just replaced the code from haraka-utils.

Reply:
Replacing the *encoding* of RFC-2047 'encoded-word's corrected the problem of including spaces. This was reflected in the test add_headers.add_utf8 where a header with a space was encoded as:
<pre><code>
'Bar: =?UTF-8?Q?b=C3=B8=C3=B8?= 1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890\n'
</code></pre>
Whereas RFC-2047 says (at the top of page 5) specifically:

<pre><code>
   IMPORTANT: 'encoded-word's are designed to be recognized as 'atom's
   by an RFC 822 parser.  As a consequence, unencoded white space
   characters (such as SPACE and HTAB) are FORBIDDEN within an
   'encoded-word'.  For example, the character sequence

      =?iso-8859-1?q?this is some text?=

   would be parsed as four 'atom's, rather than as a single 'atom' (by
   an RFC 822 parser) or 'encoded-word' (by a parser which understands
   'encoded-words').  The correct way to encode the string "this is some
   text" is to encode the SPACE characters as well, e.g.

      =?iso-8859-1?q?this=20is=20some=20text?=
</code></pre>